pages/implementers/interfaces.md

```diff
@@ -0,0 +1,203 @@
+
+#### `Instance`
+
+The type of JSON instances.
+This type should essentially be `Any` or a type capable of representing all JSON values, given JSON Schema's applicability to any representable JSON.
+
+#### `Dialect`
+
+The type of JSON dialects or dialect identifiers.
+This type may simply be `String` if the dialect is represented within the implementation by its URI or by some short name.
+
+#### `EvaluationOptions`
+
+The type of "fully ready" schemas and instances *along* with any additional implementation-specific customizable behavior.
+At minimum, this type is either `Schema → Instance` or `Schema × Instance` (depending on whether the implementation takes both schema and instance together or compiles schemas and then produces a separate function taking the instance to validate).
+It is highly likely to be richer in at least some of the following ways:
+
+  * Given likely support for some form of schema registry in order to support referencing external schemas, this type will likely include a registry of some sort, i.e. `Mapping<URI, Schema> → Schema → ...`
+  *   If an implementation supports customizing which JSON types match to which language types (such as [discussed below](#type-customization)) then this type likely includes some representation of this mapping, i.e. `Mapping<String, Callable[...]> → Schema → ...` encapsulating what each type is mapped to during this evaluation.
+  *   Similarly, if there is a specific API for [format assertion enablement](#format-assertion-enablement), some representation of the `format` keyword's behavior is present in the context
+  *   If the implementation supports the [creation or customization of dialects](#dialect-creation), and especially if schemas can contain subschemas across different dialects, then the context will contain some representation of dialects, e.g. `Dialect → Schema → ...`
+
+#### `Result`
+
+The type of JSON Schema validation results, i.e. an object which encapsulates the validity of a given `Instance` under a `Schema`.
+This type may simply be `Boolean` in some implementations or languages.
+
+`ResultWithAnnotations`
+
+> The type of JSON Schema validation results that *include* JSON Schema annotations collected during validation.
+
+#### `URI`
+
+The type of RFC 3986 compliant URIs.
+This type should not generally be the same as `String`, as URIs have multiple possible string representations and require normalization to
+have correct semantics.
+
+---
+
+Because this page deals with concerns which cross language barriers, and different programming languages have different capabilities particularly around how they represent out of band errors (via exceptions, option types, wrapper return types, sentinel values or some other mechanism), this page also introduces[^1] specific notation for representing the types of functions which include explicit representation for their out-of-band errors.
```

Comment:
I wonder if this section might be better above the definitions.  The definitions (specifically `EvaluationOptions`) use your function syntax (which this section explains), but this section doesn't use the definitions at all.

Reply:
The "new" thing in this section is meant to be the exception syntax (`<!> Foo`) not the function syntax itself -- which the definitions above don't use yet. But I'll re-read to see if this can go above.